source/h2_stream.c

```diff
@@ -791,12 +809,19 @@ int aws_h2_stream_encode_data_frame(
     if (input_stream_complete) {
         s_h2_stream_write_data_complete(stream, &waiting_writes);
     }
-
     /*
      * input_stream_complete for manual writes just means the current outgoing_write is complete. The body is not
      * complete for real until the stream is told to close
      */
     if (input_stream_complete && ends_stream) {
+        if (stream->base.on_metrics) {
+            AWS_ASSERT(stream->base.metrics.send_start_timestamp_ns != 0);
+            AWS_ASSERT(stream->base.metrics.send_end_timestamp_ns == 0);
+            aws_high_res_clock_get_ticks(&stream->base.metrics.send_end_timestamp_ns);
+            AWS_ASSERT(stream->base.metrics.send_end_timestamp_ns >= stream->base.metrics.send_start_timestamp_ns);
+            stream->base.metrics.sending_duration_ns =
+                stream->base.metrics.send_end_timestamp_ns - stream->base.metrics.send_start_timestamp_ns;
+        }
         /* Done sending data. No more data will be sent. */
```

Comment:
trivial: move this if block after the comment `/* Done sending data. No more data will be sent. */`